addons/mail/models/mail_followers.py

```diff
@@ -142,24 +142,22 @@ def _get_recipient_data(self, records, message_type, subtype_id, pids=None, cids
         LEFT JOIN res_users users ON users.partner_id = partner.id AND users.active
         LEFT JOIN res_groups_users_rel groups_rel ON groups_rel.uid = users.id
         LEFT JOIN res_groups groups ON groups.id = groups_rel.gid
-        WHERE EXISTS (
-            SELECT partner_id FROM sub_followers
-            WHERE sub_followers.channel_id IS NULL
-                AND sub_followers.partner_id = partner.id
-                AND (coalesce(sub_followers.internal, false) <> TRUE OR coalesce(partner.partner_share, false) <> TRUE)
-        ) %s
+        LEFT JOIN sub_followers subfol ON subfol.partner_id = partner.id
+        WHERE %s
+            subfol.channel_id IS NULL
+            AND (coalesce(subfol.internal, false) <> TRUE OR coalesce(partner.partner_share, false) <> TRUE)
         GROUP BY partner.id, users.notification_type
     UNION
```

Comment:
using a `UNION ALL` could also improve overall perf, without changing the result

Reply:
Done. As I understand difference is visible on massive followers of the same document, which is not the case in most of the time